### DISCUSSION TITLE
Repositories & Folders

### DISCUSSION THREAD

**Discussion by rgleason** [opencpn-manuals/main#12] [Q&A]
To start, I am confused about the structure. What are these repositories used for?
Github Account:  opencpn-manuals
- Main
- Development
- Plugins
- User

Where and how should we contribute?  

What repositories should we be using for what purposes?

For example where should I move the [21 Issues in my repository](https://github.com/rgleason/ocpn-dev-manual/issues)  to?

Should they be moved to here https://github.com/opencpn-manuals/development/issues ?

How to move the issues  Github move issues feature
https://www.google.com/search?client=firefox-b-1-m&sxsrf=ALeKk00JySG408iIU1REGKQnjqdSV6vTlw%3A1628528517631&q=github+move+issues+feature&oq=github+move+issues+feature&aqs=heirloom-srp..

After I move those issues, what do I do with https://github.com/rgleason/ocpn-dev-manual ?
Remove it?

**Comment by rgleason** [comment]
Alec wrote:
"`You will probably have to first transfer the project rgleason/ocpn-dev-manual -> opencpn-manuals/ocpn-dev-manual before you can copy things around."`

If so, that may answer some of my questions, because right now rgleason/ocpn-dev-manual is from  leamas/ocpn-dev-manual  but it is not identical, so that bothers me for some reason.

See Github Transfer 
[Github Transfer](https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository)

**Reply by rgleason** [reply]
It does not look to me like we want to transfer, it will complicate things I think.

**Reply by rgleason** [reply]
See https://github.com/opencpn-manuals/development/issues/2#issuecomment-897664635
I find can only transfer Issues within my account github.com/rgleason
So I can individually recreate the issues and summarize with critical links.
or I can link back to the Issue itself (but then I have to keep the repository intact)
What would you like?

**Reply by leamas** [reply]
I guess things are clearer now. Or?

**Reply by rgleason** [reply]
Yes

**Reply by rgleason** [reply]
Maybe I should just try it.

**Comment by rgleason** [comment]
I am confused about the structure. What are these repositories used for?
Github Account: opencpn-manuals
    Main -Parent Container
    Development  - Development Manual
    Plugins - Plugin Manual
    User - Issues related to the user manual

Where and how should we contribute?  For owners the process is different than contributors.
   Owners - Contributing to  Development Manual see [Workflow.adoc  
](https://github.com/opencpn-manuals/development/blob/main/WORKFLOWS.adoc)
   Contributing - Use the Edit button at top right corner to submit a PR (Requires github account)

What repositories should we be using for what purposes?
     Owners - for Development Manual  fork the development repository
     Still unclear how we should be editing the Plugin Manual, is it the same?
 
For example where should I move the 21 Issues in my repository to?
     Moved by hand.

Should they be moved to here https://github.com/opencpn-manuals/development/issues ?
    Moved by hand

How to move the issues Github move issues feature
https://www.google.com/search?client=firefox-b-1-m&sxsrf=ALeKk00JySG408iIU1REGKQnjqdSV6vTlw%3A1628528517631&q=github+move+issues+feature&oq=github+move+issues+feature&aqs=heirloom-srp..

After I move those issues, what do I do with https://github.com/rgleason/ocpn-dev-manual ?
Remove it?  - Yes simple remove it locally and in my remote rgleason

I don't think  I should transfer, it will just create more confusion.

Alec do you agree?

**Reply by leamas** [reply]
We are all owners. All three of us have complete permissions on all repos here.  So the term "contribute" does not really apply.

Editing the Plugin Manual is distributed, @Rasbats might need to correct me here. That said, if you clone (locally) the Plugin Manual you can make a local build. In this clone you can also modify the central layout, but it might be better to leave this to Mike for now. 

However, you need to look into the sources.state file. Here you can see a long list, an excerpt here: 
```
/radar_pi https://github.com/opencpn-radar-pi/radar_pi.git e3cf4a3e0c0d83b2d22adc8ac8ea8e700f3b42c5
./aisradar_pi https://github.com/nohal/aisradar_pi.git b3f8dc395b7bfda22c3543e8aa86faae31d0678f
./rtlsdr_pi https://github.com/rgleason/rtlsdr_pi.git ba07e036f00264c299378423a9a0db10b334cf69
./nvc_pi https://github.com/bdbcat/nvc_pi.git e6950f2c322eca9b4397c3bfabd990789d3e2255
./s63_pi https://github.com/bdbcat/s63_pi.git 1809b1744f5d1af0824cd64c18c20862258fedcb
./bsb4_pi https://github.com/bdbcat/bsb4_pi.git 3312414893f20642c6a58d334716a052f6d29c14
./oeSENC_pi https://github.com/bdbcat/oeSENC_pi.git adc128678ff2b29a2eb25016aa51c115895d2822
```

From this list you can see that if you want to modify for example rtlsdr you need to clone  https://github.com/rgleason/rtlsdr_pi.git (locally), and push changes onto it. Those changes will be visible next time you build the Plugins Manual, but you can of course also make a local build in the plugin (for example rtlsdr) to check results.

If you want to modify a repo which isn't yours you need to make a PR, but I guess this is basically out of scope for now.

**Comment by leamas** [comment]
> After I move those issues, what do I do with https://github.com/rgleason/ocpn-dev-manual ?

You can just remove it, it adds nothing but confusion to the show.

**Comment by leamas** [comment]
> How to move the issues Github move issues feature

For this to work, you mist first transfer your repo rgleason/ocpn-dev-manual from your personal user rgleason to the opencpn organisation (which you are member of). When this is done, you can just click the transfer button which is somewhere middle-right when looking at an issue.

**Comment by rgleason** [comment]
Thanks Alec, that's helpful.

**Comment by leamas** [comment]
tl:dr When you should edit the manual for a plugin, you work in the plugin. As an example, if you want to update rtlsdr_pi you work in that plugin. Mike has added a manual/ directory where the plugin manual lives.

Besides the link limitations, you can preview changes using firefox. 

The overall Plugins Manual "sees" your changes when you push them the same way as you would push a change to the code.

**Comment by leamas** [comment]
If you want to make a local build, add a file _manual/rtlsdr/site.yml_ which looks like this:

```
site:
  title: rtlsdr_pi Manual
  url:  https://nowhere.net
  start_page: rtlsdr:ROOT:rtlsdr.adoc
content:
  sources:
    - url: ../..
      start_path: manual/rtlsdr/
      branches: HEAD
      edit_url: true
ui:   
  bundle:
    url: https://gitlab.com/leamas/antora-ui-default/-/raw/master/latest/ui-bundle.zip
    snapshot: true
output:
  dir: ./docs
  clean: true              
```

You can then build the docs using `antora site.yml` and check results by pointing browser to _docs/index.html_
